WooCommerce/Classes/Authentication/Epilogue/StorePickerCoordinator.swift

```diff
@@ -35,16 +40,35 @@ final class StorePickerCoordinator: Coordinator {
         return pickerVC
     }()
 
-    init(_ navigationController: UINavigationController, config: StorePickerConfiguration) {
+    init(_ navigationController: UINavigationController,
+         config: StorePickerConfiguration,
+         switchStoreUseCase: SwitchStoreUseCaseProtocol = SwitchStoreUseCase(stores: ServiceLocator.stores)) {
         self.navigationController = navigationController
         self.selectedConfiguration = config
+        self.switchStoreUseCase = switchStoreUseCase
     }
 
     func start() {
         showStorePicker()
     }
-}
 
+    /// Checks available stores and switch store if only one valid store is available
+    ///
+    @MainActor
+    func switchStoreIfOnlyOneStoreIsAvailable() async throws {
+        let sites = switchStoreUseCase.getAvailableStores()
+
+        guard let siteID = sites.first?.siteID else {
+            throw StoreSelectionError.noStoresAvailable
```

Comment:
❓ the condition just checks for the site ID existence, while the error is "no stores available." what do you think about checking `sites.isNotEmpty` here?

Reply:
I needed the siteID below.
Refactored in in https://github.com/woocommerce/woocommerce-ios/pull/9444/commits/4aa04057c7c6ca5346c1a99e756a58641c027fd6